base/sort.jl

```diff
@@ -1464,10 +1464,12 @@ import Core.Intrinsics: slt_int
 import ..Sort: sort!, UIntMappable, uint_map, uint_unmap
 import ...Order: lt, DirectOrdering
 
-const Floats = Union{Float32,Float64}
-const FPSortable = Union{ # Mixed Float32 and Float64 are not allowed.
+const Floats = Union{Float16, Float32, Float64} # IEEEFloat is not available in Core.Compiler
+const FPSortable = Union{ # Mixed bitwidths are not allowed.
+    AbstractVector{Union{Float16, Missing}},
     AbstractVector{Union{Float32, Missing}},
     AbstractVector{Union{Float64, Missing}},
+    AbstractVector{Float16},
```

Comment:
maybe `AbstractVector{<:IEEFloat}`

Reply:
That would include Vector{Union{Float32, Float64}} which is not safe to compare via `slt_int` (see https://github.com/JuliaLang/julia/issues/42739). I'll link to that issue in a comment.

Reply:
makes sense.